### PR TITLE
Implement the CCM/CSI migration

### LINKED ISSUE #1451: Implement the base "migrate ccm-csi" command
This issue assumes implementing the `migrate ccm-csi` command which will be used to trigger the CCM/CSI migration. The command will have logic/tasks that are common for all providers. Later on, the command will be extended with the logic/tasks specific to the cloud provider.

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

This PR is a follow-up of #1468. The following has been done in this PR:

* Improve stability of the CCM migration process implemented in #1468 by:
  * Verifying that both kube-apiserver and kube-controller-manager are ready before updating Kubelet config and restarting it
  * Verifying that Kubelet has successfully restarted before proceeding to the next node
* Implement the CSI migration process
  * This assumes applying the appropriate feature gates on kube-apiserver, kube-controller-manager, and kubelet, as well as, deploying the CSI plugin
* Implement the `--complete` flag used to complete the CCM/CSI migration
  * The CCM/CSI migration process is done in two phases, where:
    * Phase 1 assumes disabling cloud-related in-tree controllers in favor of external CCM. The volume controllers are still left enabled until the worker nodes are not rolled out
    * Phase 2 assumes fully disabling the in-tree cloud provider

Known limitations and issues:

* The CCM/CSI migration currently works only for OpenStack
  * vSphere support will be added in a follow-up PR (#1471 tracks vSphere support)
  * Support for other providers will be added in future KubeOne releases
* The OpenStack CCM/CSI migration document states that PersistentVolumes must be annotated, so the CSI driver can properly clean them up. This part isn't done. It's still unclear can this be automated, so I'll leave it for a follow up (#1452 tracks this issue)
* The migration process has been tested only with Kubernetes 1.22.1. I'll create a dedicated Epic for in-depth testing of this feature.

On top of that, this PR also removes the kubeadm v1beta1, which has been used for Kubernetes clusters up to 1.15. Since we're not supporting clusters older than 1.19 (#1466), it doesn't make much sense to maintain the unused API/code.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1451
Relevant to #1452

**Does this PR introduce a user-facing change?**:
```release-note
Implement the CCM/CSI migration. The migration is currently available for OpenStack clusters. Support for other cloud providers will be added in future KubeOne releases. Make sure to familiarize yourself with the migration process before getting started by checking the command description.
```

/assign @kron4eg

### CODE REVIEW

File: pkg/apis/kubeone/helpers.go [hunk 1]
```diff
@@ -153,6 +154,60 @@ func (p CloudProviderSpec) CloudProviderInTree() bool {
...
+// enabled by default since Kubernetes 1.18
+// (https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/)
+// This is a KubeOneCluster function because feature gates are Kubernetes-version dependent.
+func (c KubeOneCluster) CSIMigrationFeatureGates(complete bool) (map[string]bool, string, error) {
+	if c.CloudProvider.Openstack != nil {
+		featureGates := map[string]bool{
+			"CSIMigrationOpenStack": true,
+			"ExpandCSIVolumes":      true,
+		}
+		featureGatesString := "CSIMigrationOpenStack=true,ExpandCSIVolumes=true"
```
[the remark this answers is not in the document]
Comment (xmudrii) [reply pkg/apis/kubeone/helpers.go]: Fixed, PTAL.

File: pkg/apis/kubeone/helpers.go [hunk 2]
```diff
@@ -208,6 +207,24 @@ func (c KubeOneCluster) InTreePluginUnregisterFeatureGate() string {
...
+		if !first {
+			featureGatesStr += ","
+		} else {
+			first = false
+		}
+		featureGatesStr += k + "=" + strconv.FormatBool(v)
+	}
+
+	return featureGatesStr
+}
```
Comment (kron4eg) [inline pkg/apis/kubeone/helpers.go:226]:
```
func marshalFeatureGates(fgm map[string]bool) string {
	keys := []string{}
	for k, v := range fgm {
		keys = append(keys, fmt.Sprintf("%s=%t", k, v))
	}

	sort.Strings(keys)
	return strings.Join(keys, ",")
}
```

Comment (xmudrii) [reply pkg/apis/kubeone/helpers.go:219]: Nice trick. Updated, PTAL.

**Review by kron4eg — APPROVED** [review]

**Pull request merged**